vars/cloudFoundryDeploy.groovy

```diff
@@ -196,11 +196,19 @@ private String deleteOptionIfRequired(Map config) {
     }
 }
 
-private String stopOldAppIfRequired(Map config) {
+private void stopOldAppIfRequired(Map config) {
+    String oldAppName = "${config.cloudFoundry.appName}-old"
+    String cfStopOutputFileName = 'cfStopOutput.txt'
+
     if (config.keepOldInstance && config.deployType == 'blue-green') {
-        return "cf stop ${config.cloudFoundry.appName}-old"
-    } else {
-        return ''
+        int cfStopReturncode = sh (returnStatus: true, script: "cf stop $oldAppName  > $cfStopOutputFileName")
+
+        if (cfStopReturncode > 0) {
+            String cfStopOutput = readFile(file: cfStopOutputFileName)
```

Comment:
Waiting for an error, parsing log files for certain entries seems only to be second best approach to me (e.g. log file formats are normally not considered to be API and might be changed from one version to another).

Sometimes something like this is required though since there are no other reasonable options. Is there any command available which can be used for checking if an application with a certain name is deployed/running? If so I would like to suggest to use such an approach.

Reply:
I could use cf app `appName-old` which returns 1 if there is no app and log would look as follows:
>Showing health and status for app address-manager-old1 in org XXXX / space XXX as XXX

>App address-manager-old1 not found
>FAILED

Nevertheless I would prefer the current implementation because with `cf app appName-old` I would still parse the log in case of rc > 0.